File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/watch/execution_watch_resync_required.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_watch_resync_reason import AiSdkExecutionWatchResyncReason


@dataclass(frozen=True)
class AiSdkExecutionWatchResyncRequired(AiSdkWireModel):
    reason: AiSdkExecutionWatchResyncReason
    schema_version: int = AiSdkSchemaVersions.EXECUTION_WATCH_RESYNC_REQUIRED
    requested_after_sequence: int | None = None
    earliest_available_sequence: int | None = None
    latest_sequence: int | None = None
    message: str | None = None

File: implementations/node/sdk/src/contracts/watch/execution-watch-resync-required.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionWatchResyncReason } from "./execution-watch-resync-reason.js";

export interface AiSdkExecutionWatchResyncRequired {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionWatchResyncRequired;
  readonly reason: AiSdkExecutionWatchResyncReason;
  readonly requestedAfterSequence?: number;
  readonly earliestAvailableSequence?: number;
  readonly latestSequence?: number;
  readonly message?: string;
}
